crates/adapters/src/transport/durable/mod.rs

```diff
@@ -0,0 +1,192 @@
+//! Durable input for fault tolerance.
+//!
+//! Fault tolerance requires that we be able to divide input into numbered steps
+//! that can be retrieved repeatedly with the same content, despite crashes.
+//! [`DurableInputTransport`] and [`DurableInputEndpoint`] are analogs to
+//! [`InputTransport`] and [`InputEndpoint`], respectively, that provide
+//! durability.
```

Comment:
maybe also a sentence to explain high-level the reader and consumer in this file and how they relate

Reply:
OK, I added this:

```patch
diff --git a/crates/adapters/src/transport/durable/mod.rs b/crates/adapters/src/transport/durable/mod.rs
index cdde88cb..66458394 100644
--- a/crates/adapters/src/transport/durable/mod.rs
+++ b/crates/adapters/src/transport/durable/mod.rs
@@ -11,6 +11,25 @@
 //! * `kafka`, for input from [Kafka](https://kafka.apache.org) via
 //! [`KafkaDurableInputTransport`].
 //!
+//! # Types
+//!
+//! The types in this module have the following relationships:
+//!
+//! * [`DurableInputTransport`] takes a name and a configuration to yield a
+//!   [`DurableInputEndpoint`].
+//!
+//! * A [`DurableInputEndpoint`] can do a couple of basic operations on an
+//!   instance of durable input.  Most notably, it can open it starting at a
+//!   particular step.  The open method takes a [`DurableInputConsumer`] as
+//!   a parameter and returns a [`DurableInputReader`].
+//!
+//! * [`DurableInputReader`] allows the client to request steps to read or
+//!   append.
+//!
+//! * The client provides a [`DurableInputConsumer`] when it opens an endpoint.
+//!   The reader asynchronously calls its method to report all of the data and
+//!   status information from reading a durable input.
+//!
 //! [`InputTransport`]: crate::transport::InputTransport
 //! [`InputEndpoint`]: crate::transport::InputEndpoint
 
```